modules/katello_manifest.py

```diff
@@ -0,0 +1,196 @@
+author: "Andrew Kofink (@akofink)"
+requirements:
+    - "nailgun >= 0.29.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+    organization:
+        description:
+            - Organization that the sync plan is in
+        required: true
+    manifest_path:
+        description:
+            - Path to the manifest zip file
+        required: true
+    state:
+        description:
+            - The state of the manifest
+        required: true
+        choices:
+            - absent
+            - present
+            - refreshed
+    repository_url:
+        description:
+            - URL to retrieve content from
+        default: https://cdn.redhat.com
+'''
+# HERE
+EXAMPLES = '''
+- name: "Upload the RHEL developer edition manifest"
+  katello_manifest:
+    username: "admin"
+    password: "admin"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    organization: "Default Organization"
+    state: present
+    manifest_path: "/tmp/manifest.zip"
+  delegate_to: localhost
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+        Subscription
+    )
+
+    from nailgun.entity_mixins import (
+        TaskFailedError
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.ansible_nailgun_cement import (
+    create_server,
+    ping_server,
+    find_organization,
+    current_subscription_manifest,
+    set_task_timeout,
+)
+
+
+def validate_params(module, state=None, manifest_path=None):
+    manifest_supplied = not (manifest_path is None or len(manifest_path) == 0)
+    if state == 'present' and not manifest_supplied:
+        module.fail_json(msg="manifest_path is required when state is present.")
+    elif state == 'absent' and manifest_supplied:
+        module.fail_json(msg="manifest_path is prohibited when state is absent.")
+    elif state == 'refreshed' and manifest_supplied:
+        module.fail_json(msg="manifest_path is prohibited when state is refreshed.")
+
+
+def manifest(module, organization, state, manifest_path=None, repository_url="https://cdn.redhat.com"):
+    changed = False
+    organization = find_organization(module, organization).read()
+    current_manifest = current_subscription_manifest(module, organization)
+    manifest_present = current_manifest is not None
+
+    if organization.redhat_repository_url != repository_url:
+        organization.redhat_repository_url = repository_url
+        organization.update({'redhat_repository_url'})
+        changed = True
+
+    if manifest_path is not None and state == 'present':
```

Comment:
is the `manifest_path` check required here? `validate_params` should have already caught that?